From 5ac8e03bdccc31bb4f936b202448466650330690 Mon Sep 17 00:00:00 2001
From: Neeraj Singh <neerajdotname@gmail.com>
Date: Sun, 26 Sep 2010 00:54:44 -0400
Subject: [PATCH] Fix for #5579 involved the code change for both has_one and has_many
 relationships. The path included test only for has_one. This patch
 adds test for has_many relationship.

[#5706 state:resolved]
---
 activerecord/test/cases/nested_attributes_test.rb |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/activerecord/test/cases/nested_attributes_test.rb b/activerecord/test/cases/nested_attributes_test.rb
index 75ffd31..54429bf 100644
--- a/activerecord/test/cases/nested_attributes_test.rb
+++ b/activerecord/test/cases/nested_attributes_test.rb
@@ -115,7 +115,7 @@ class TestNestedAttributesInGeneral < ActiveRecord::TestCase
     assert_difference('Ship.count') { pirate.save! }
   end
 
-  def test_reject_if_with_a_proc_which_returns_true_always
+  def test_reject_if_with_a_proc_which_returns_true_always_for_has_one
     Pirate.accepts_nested_attributes_for :ship, :reject_if => proc {|attributes| true }
     pirate = Pirate.new(:catchphrase => "Stop wastin' me time")
     ship = pirate.create_ship(:name => 's1')
@@ -123,6 +123,14 @@ class TestNestedAttributesInGeneral < ActiveRecord::TestCase
     assert_equal 's1', ship.reload.name
   end
 
+  def test_reject_if_with_a_proc_which_returns_true_always_for_has_many
+    Man.accepts_nested_attributes_for :interests, :reject_if => proc {|attributes| true }
+    man = Man.create(:name => "John")
+    interest = man.interests.create(:topic => 'photography')
+    man.update_attributes({:interests_attributes => { :topic => 'gardening', :id => interest.id } })
+    assert_equal 'photography', interest.reload.topic
+  end
+
  def test_has_many_association_updating_a_single_record
     Man.accepts_nested_attributes_for(:interests)
     man = Man.create(:name => 'John')
-- 
1.7.2

